packages/ui/src/forum/components/Thread/WatchlistButton.tsx

```diff
@@ -13,7 +13,7 @@ export const WatchlistButton = ({ threadId }: Props) => {
   const [watchlist, setWatchlist] = useLocalStorage<string[]>(FORUM_WATCHLIST)
   const addToWatchlist = useCallback(() => setWatchlist(watchlist ? [...watchlist, threadId] : [threadId]), [watchlist])
   const removeFromWatchlist = useCallback(
-    () => setWatchlist(watchlist?.filter((id) => id !== threadId) ?? []),
```

Comment:
@traumschule currently this code: `setWatchlist(watchlist?.filter((id) => id !== threadId) ?? []),` is fine, it works just as well as `setWatchlist(watchlist ? [...watchlist, threadId].filter((id) => id !== threadId) : [])` the issue I mentioned in #3769 is the value of `watchlist` comes from the local storage state at the moment the component was mounted. It's fine in most cases but if someone had different threads opened in different browser tabs/windows: if 2 threads are added in 2 different tabs/windows the watchlist will only have the latest addition.

A similar problem caused #3711 fixed here: #3719 by mimicking the `useState` api (i.e when providing a function perform an updated) except that update function are provided with an up to date value from the local storage. But this PR was merged in `carthage` this is why we should wait for `carthage` to be merged into `dev` to fix this.

I'm closing this PR in the meantime.

Reply:
Sounds good and sorry for the extra work! I'll keep this as low-prio draft on my end.